src/app/rpmostree-override-builtins.cxx

```diff
@@ -89,7 +93,64 @@ handle_override (RPMOSTreeSysroot  *sysroot_proxy,
   g_autoptr(GVariant) options = g_variant_ref_sink (g_variant_dict_end (&dict));
 
   g_autoptr(GVariant) previous_deployment = rpmostree_os_dup_default_deployment (os_proxy);
-
+  g_autoptr(GUnixFDList) fetched_rpms_fds = NULL;
+  g_autoptr(GPtrArray) override_replace_final = NULL; 
+  if (override_replace && opt_ex_pin_from_repos)
+    {
+      override_replace_final = g_ptr_array_new_with_free_func (free);
+      g_autoptr(GPtrArray) queries = g_ptr_array_new ();
+
+      for (const char *const* it = override_replace; it && *it; it++)
+        {
+          auto pkg = *it;
+
+          if (rpmostreecxx::is_http_arg (pkg) || rpmostreecxx::is_rpm_arg (pkg))
+            {
+              g_ptr_array_add (override_replace_final, (gpointer)g_strdup (pkg));
+              continue;
+            }
+          else
+            {
+              g_ptr_array_add (queries, (gpointer)pkg);
+            }
+        }
+      if (queries->len > 0)
+        {
+          g_ptr_array_add (queries, NULL);
+          g_autofree char *refresh_transaction_address = NULL;
+
+          if (!rpmostree_os_call_refresh_md_sync (os_proxy, options, &refresh_transaction_address,
+                                                 cancellable, error))
+            return FALSE;
+          if (!rpmostree_transaction_get_response_sync (sysroot_proxy, (const char*)refresh_transaction_address,
+                                                       cancellable, error))
+            return FALSE;
+          if (!rpmostree_osexperimental_call_download_packages_sync (osexperimental_proxy,
+                                                                     (const char *const*)queries->pdata,
+                                                                     NULL, &fetched_rpms_fds,
+                                                                     cancellable, error))
+            return FALSE;
+
+          const int nfds = fetched_rpms_fds ? g_unix_fd_list_get_length (fetched_rpms_fds) : 0;
+
+          if (nfds == 0)
+            {
+              g_print ("No matching package names at enabled repositories\n");
```

Comment:
Hmm, I don't think this can happen in practice because the D-Bus method should now correctly fail if no packages are found. But good to still assert at least. So maybe just something like

```c
if (!rpmostree_osexperimental_call_download_packages_sync (...))
  return FALSE;

const int nfds = fetched_rpms_fds ? g_unix_fd_list_get_length (fetched_rpms_fds) : 0;
g_assert_cmpuint (nfds, >, 0);
for (guint i = 0; i < nfds, i++)
  {
    ...
  }
```

?